FlowCal/io.py

```diff
@@ -171,14 +172,18 @@ def read_fcs_text_segment(buf, begin, end, delim=None, supplemental=False):
                              + " TEXT segment")
         else:
             buf.seek(begin)
-            delim = str(buf.read(1))
+            delim = buf.read(1)
+            if sys.version_info.major >= 3:
+                delim = chr(delim[0])
```

Comment:
`buf.read()` produces a byte (a number), which needs to be casted to a string for downstream use. See my comment in the "Conversation" tab.